COMETwebapp/COMETwebapp/Components/OpenModel.razor

```diff
@@ -0,0 +1,194 @@
+    private List<DomainOfExpertise> DomainList { get; set; } = new List<DomainOfExpertise>();
+
+    /// <summary>
+    /// The <see cref="EngineeringModelSetup"/> selected
+    /// </summary>
+    private EngineeringModelSetup ModelSetup { get; set; }
+
+    /// <summary>
+    /// The <see cref="IterationSetup"/> selected
+    /// </summary>
+    private IterationSetup IterationSetup { get; set; }
+
+    /// <summary>
+    /// Initialize <see cref="ModelSetupList"/> according to the opened Session and the <see cref="ActivePerson"/>
+    /// </summary>
+    private void InitializeModelList()
+    {
+        foreach (var model in ISessionAnchor.Session.RetrieveSiteDirectory().Model)
+        {
+            foreach (var participant in model.Participant)
+            {
+                if (participant.Person.Name.Equals(ISessionAnchor.Session.ActivePerson.Name))
+                {
+                    this.ModelSetupList.Add(model);
+                }
+            }
+        }
+    }
+
+    /// <summary>
+    /// Initialize <see cref="DomainList"/> and <see cref="IterationSetupList"/> according to the selected <see cref="ModelSetup"/>
+    /// </summary>
+    /// <param name="e"> Event emmited when a ModelSetup is selected </param>
+    private void SetNameModel(ChangeEventArgs e)
+    {
+        foreach (var model in ISessionAnchor.Session.RetrieveSiteDirectory().Model)
+        {
+            if (model.Name.Equals(e.Value.ToString()))
+            {
+                this.ModelSetup = model;
+                foreach (var participant in model.Participant)
+                {
+                    if (participant.Person.Name.Equals(ISessionAnchor.Session.ActivePerson.Name))
+                    {
+                        foreach(var domain in participant.Domain)
+                        {
+                            this.DomainList.Add(domain);
+                        }
+                    }
+                }
+                foreach(var iteration in model.IterationSetup)
+                {
+                    IterationSetupList.Add(iteration);
+                }
+            }
+        }
+    }
+
+    /// <summary>
+    /// Set the <see cref="CurrentDomainOfExpertise"/> of the <see cref="ISessionAnchor"/>
+    /// </summary>
+    /// <param name="e">Event emitted when a domain is selected</param>
+    private void SetDomain(ChangeEventArgs e)
+    {
+        foreach (var domain in this.DomainList)
+        {
+            if (domain.Name.Equals(e.Value.ToString()))
+            {
+                this.ISessionAnchor.CurrentDomainOfExpertise = domain;
+            }
+        }
+    }
+
+    /// <summary>
+    /// Set the <see cref="IterationSetup"/>
+    /// </summary>
+    /// <param name="e">Event emitted when an iteration is selected</param>
+    private void SetIteration(ChangeEventArgs e)
+    {
+        foreach (var iteration in this.IterationSetupList)
+        {
+            if (iteration.Description.Equals(e.Value.ToString()))
```

Comment:
the selection should be done on iteration number i think, not on description. Potentially you can state the number and the date? (please have a look what it looks like in the IME and repeat that, so we keep the user experience close)